src/kademlia/dht_storage.cpp

```diff
@@ -154,15 +154,15 @@ namespace
 	// less important to keep
 	struct immutable_item_comparator
 	{
-		immutable_item_comparator(node_id const& our_id) : m_our_id(our_id) {}
+		immutable_item_comparator(std::vector<node_id> const& node_ids) : m_node_ids(node_ids) {}
 		immutable_item_comparator(immutable_item_comparator const& c)
-			: m_our_id(c.m_our_id) {}
+			: m_node_ids(c.m_node_ids) {}
 
 		bool operator() (std::pair<node_id, dht_immutable_item> const& lhs
 			, std::pair<node_id, dht_immutable_item> const& rhs) const
 		{
-			int l_distance = distance_exp(lhs.first, m_our_id);
-			int r_distance = distance_exp(rhs.first, m_our_id);
+			int l_distance = min_distance_exp(lhs.first, m_node_ids);
+			int r_distance = min_distance_exp(rhs.first, m_node_ids);
 
 			// this is a score taking the popularity (number of announcers) and the
```

Comment:
I see, this isn't distance really, it's score.. so maybe the lowest score is the correct one then